OutOfSchool/OutOfSchool.Redis/CacheService.cs

```diff
@@ -74,48 +74,25 @@ IOptions<RedisConfig> redisConfig
         return returnValue;
     }
 
-    public Task SetAsync<T>(
-        string key,
-        T value,
-        TimeSpan? absoluteExpirationRelativeToNowInterval = null,
-        TimeSpan? slidingExpirationInterval = null)
-        => ExecuteRedisMethod(() => {
-            var options = new DistributedCacheEntryOptions
-            {
-                AbsoluteExpirationRelativeToNow = absoluteExpirationRelativeToNowInterval ?? redisConfig.AbsoluteExpirationRelativeToNowInterval,
-                SlidingExpiration = slidingExpirationInterval ?? redisConfig.SlidingExpirationInterval
-            };
-
+    public async Task RemoveAsync(string key)
+        => await ExecuteRedisMethod(async () => {
             cacheLock.EnterWriteLock();
             try
             {
-                cache.SetString(key, JsonConvert.SerializeObject(value), options);
-            }
-            finally
-            {
-                cacheLock.ExitWriteLock();
-            }
-        });
-        
-    public Task ClearCacheAsync(string key)
-        => ExecuteRedisMethod(() => {
-            cacheLock.EnterWriteLock();
-            try
-            {
-                cache.Remove(key);
+                await cache.RemoveAsync(key);
             }
             finally
             {
                 cacheLock.ExitWriteLock();
             }
         });
 
-    public Task RefreshAsync(string key)
-        => ExecuteRedisMethod(() => {
+    public async Task RefreshAsync(string key)
+        => await ExecuteRedisMethod(async () => {
```

Comment:
Please avoid adding excessive async/await. Your code could be simplified to
```
    public Task RefreshAsync(string key)
        => ExecuteRedisMethod(async () => {
            ...
```

Reply:
Deleted async/await